examples/internal/proto/examplepb/echo_service.pb.gw.go

```diff
@@ -231,12 +231,15 @@ func request_EchoService_Echo_2(ctx context.Context, marshaler runtime.Marshaler
 		return nil, metadata, status.Errorf(codes.InvalidArgument, "missing parameter %s", "lang")
 	}
 
-	if protoReq.Code == nil {
-		protoReq.Code = &SimpleMessage_Lang{}
-	} else if _, ok := protoReq.Code.(*SimpleMessage_Lang); !ok {
-		return nil, metadata, status.Errorf(codes.InvalidArgument, "expect type: *SimpleMessage_Lang, but: %t\n", protoReq.Code)
-	}
-	protoReq.Code.(*SimpleMessage_Lang).Lang, err = runtime.String(val)
+	func() *SimpleMessage_Lang {
+		msg := &protoReq
+		oneof := msg.Code
+		if _, ok := oneof.(*SimpleMessage_Lang); oneof == nil || !ok {
+			oneof = new(SimpleMessage_Lang)
+			msg.Code = oneof
+		}
+		return oneof.(*SimpleMessage_Lang)
+	}().Lang, err = runtime.String(val)
```

Comment:
This looks super whack. Do we need to put this in its own little function like this? Can we not do this in the outer scope?

Reply:
Agreed. It was the simplest way I could find to cover all cases. 

Having this function wrapping everything makes the oneof template also work for cases like [` request_EchoService_EchoBody_1` below](https://github.com/grpc-ecosystem/grpc-gateway/pull/2739/files#diff-72aeb6791cd8a6135693f8392966b173654d28b572b94d68f84796046540dd2cR562).

Do you think a better solution would involve editing the templates directly? I've been playing around with such an approach but nothing that works quite yet with minimal impact on the current templates.

Reply:
I'm not sure what you mean by editing the templates directly, but we certainly should consider making more sweeping changes to fit in this behavior. Obviously the less we can touch the better.

It also looks like you need to format the protobuf files, you can just run `buf format -w` and that should do it.

Reply:
No worries. I think we have to go back to the drawing board and accept that there will be template changes to support this. I haven't really got the bandwidth to work on this right now but will be happy to review any ideas you might have.

Reply:
Ok cool. Let me dig a bit more into the code and see what I manage to come up with. I'll mark this as draft for now

Reply:
@johanbrandhorst ended up going on a different route (similar to the old #570). Looks much better now, I hope.